packages/beacon-node/src/api/impl/lodestar/index.ts

```diff
@@ -187,6 +190,33 @@ export function getLodestarApi({
     async dumpDbStateIndex() {
       return {data: await db.stateArchive.dumpRootIndexEntries()};
     },
+    async getHistoricalSummaries({stateId}, _context) {
+      const {state} = await getStateResponseWithRegen(chain, stateId);
+      let slot: number;
+      if (state instanceof Uint8Array) {
+        slot = getStateSlotFromBytes(state);
+      } else {
+        slot = state.slot;
+      }
+      if (config.getForkSeq(slot) < ForkSeq.capella) {
+        throw new Error("Historical summaries are not supported before Capella");
+      }
+      const fork = config.getForkName(slot) as Exclude<ForkName, "phase0" | "altair" | "bellatrix">;
+
+      const stateView =
+        state instanceof Uint8Array ? loadState(config, chain.getHeadState(), state).state : state.clone();
```

Comment:
```suggestion
        (state instanceof Uint8Array ? loadState(config, chain.getHeadState(), state).state : state.clone()) as BeaconStateCapella;
```

You can avoid the `any` below by casting this to a `BeaconStateCapella` here

Reply:
It seems like the proof size for `HistoricalSummaries` changed in `deneb` (and will maybe again in Electra).  Is it still safe to cast it to `BeaconStateCapella` here since the shape of the state is changing somewhat?